coordinator/scripts/luigi_scoop_server.py

```diff
@@ -0,0 +1,206 @@
+    
+    # Status as recived from scooping_planning/scoop
+    self.scooping_status_ = 'NULL'
+
+    callback_lambda = lambda x: self.RecieveIceCreamOrder_(x)
+    self.server_ = actionlib.SimpleActionServer('luigi_scoop', OrderIceCreamAction, self.RecieveIceCreamOrder_, auto_start=False)
+    self.server_.start()
+
+  # Callback for sub of topic "scooping_planning/status"
+  def ScoopStatusCallback(self, scooping_status):
+    self.scooping_status_ = scooping_status
+    print(scooping_status)
+
+  # Arguments are of Pose type
+  def TranslationalPTPMotionClient(self, startPosition, endPosition):
+    rospy.wait_for_service('scooping_planning/scoop')
+    try:
+      # TODO: Only call this when status is IDLE
+      TranslationalPTPMotionServClient = rospy.ServiceProxy('scooping_planning/scoop', TranslationalPTPMotion)
+      resp = TranslationalPTPMotionServClient(startPosition, endPosition)
+      return resp.success
+    except rospy.ServiceException, e:
+      print "Service call failed: %s"%e
+
+  def SendFeedbackLuigi(self, feedback):
+    for i in scoops:
+      self._feedback.finished_scoops.append(0)
+    self._feedback.status_message = 'This is feedback'
+    # publish the feedback 
+    self.server_.publish_feedback(self._feedback)
+
+
+  # For now we set which flavor is where by hand (manually), unless vision can provide us with an api
+  def GoToFlavor(self, flavor):
+    leftStartPoint = Point(x=-0.1, y=-0.4, z=0.25)
+    rightStartPoint = Point(x=-0.1, y=-0.4, z=0.25)
+    startingPoint = Point()
+
+    # Left ice cream box
+    if flavor == availableFlavors[0]:
+      startingPoint = leftStartPoint
+    elif flavor == availableFlavors[1]:
+      startingPoint = rightStartPoint
+    # Do we fancy a third ice cream box?
+    else:
+      startingPoint = leftStartPoint
+
+    return startingPoint
+
+    
+  def RecieveIceCreamOrder_(self, data):
+
+    # success of the total scoping operation
+    success = False
+
+    # Scoops required
+    scoops = data.scoops
+
+    # Scoops done
+    scoopsScooped = np.zeros(len(scoops))
+
+    # flavors for each "scoops" ordered, not scoop :D
+    flavors = data.flavors
+
+    self.scooping_human_status_ = 'Recived a new ice cream order'
+    # intially no scoops has been scooped
+    for i in scoops:
+      self._feedback.finished_scoops.append(0)
+
+    scoops = data.scoops
+    flavors = data.flavors
+
+    r = rospy.Rate(1)
+
+    # start executing the action
+    # check that preempt has not been requested by the client
+    if self.server_.is_preempt_requested():
+      rospy.loginfo('%s: Preempted' % self._action_name)
+      self.server_.set_preempted()
+      success = False
+
+    # Go to the start point, Arne should give this point from simulation
+    startPoint = Point(x=-0.1, y=-0.4, z=0.25)
+
+    # TODO: Check the successs of the return
+    scoopingResponse = self.TranslationalPTPMotionClient(startPoint, startPoint)
+
+    # success of the total scoping operation
```

Comment:
Nope. As we discusses, the return value of the scooping service is not the overall scooping success. 
Success is true when the starting of the scooping action succeeded, false else.

Reply:
Oh this success is used for sending the final result to luigi, not the response from the scooping service

Reply:
alright!